### PR TITLE
Add endpoints to readme #23

### LINKED ISSUE #23: Create Endpoint Documentation
- [x] Add the following Endpoints to BE's readme
**note:** `Phases` correlate to FE's plan to implement, if building these endpoints in a different order makes more sense, go for it! 
#### Phase 1: Employers search and view users
- [x] All users (as an array, see FE-11 for what data we're looking for) 
  * doesn't need messages
  * allows us to test connectivity easily before designing intense logic
  * load all users to design search-results page (and applicantPreview components)
- [x] search endpoint (search => searchResults)
  * Query params or body info? 
  * doesn't need messages
  * allows us to test connectivity easily before designing intense logic
  * load all users to design search-results page (and applicantPreview components)
- [x] applicant search-options (search component onMount)
  * Returns an object with `skills` and `values` arrays
  * arrays include all attributes present on user profiles
#### Phase 2: User creation 
- [x] Single user (myAccount (in demo, after account has been created)
  * for "my account", 
    * #11 with personal info too 
    *`assuming auth`, find by username or email, returns everything in a profile
  * or for when you click a search result, although we may not need that
- [x] post new users (myAccount => create user form (post) => my profile page)
  * This will be triggered at the start of clicking "My Profile" in demo mode
- [x] post changes to a single user (myAccount => edit) *extension*
  * Allows a user to edit their own profile 
  * Since they're logged in, they can use the ID
#### Phase 3: Messaging Users
- [x] all messages belonging to user by `id`
  * (may not be necessary given the `Single User` path)
- [x] post new message
  * Any user can do this, contains the id of the applicant receiving the message

### PR DESCRIPTION
**Pull request by GreysonElkins**
#### Type of Change Made 
- [x] documentation 
#### What's this PR do?
Provides examples of data to be sent and recieved on fetch to specific endpoints
#### Where should the reviewer start?
ReadMe is the only place you can possibly start
#### How should this be manually tested?
How do you test `markdown`? Someone teach me
#### Any background context you want to provide?
These are organized by path, not by order of importance. The Front End's plan for implementation is outlined in #23 but can start with `dummy` fetches and data, so whatever y'all's workflow ends up being is good by us! (extensions other than login are included here)
#### What are the relevant tickets?
close #23

### CODE REVIEW

File: README.md
Comment (amykarnaze) [inline README.md:41]: I like the bio addition

File: README.md
Comment (ErinUntermeyer) [inline README.md:72]: Is this logic going to live on the FE and BE?

File: README.md
Comment (amykarnaze) [inline README.md:104]: That's a good looking object

File: README.md
Comment (ErinUntermeyer) [inline README.md]: Oops! I think this was meant to go where timestamp is on like 97 and have a fake username here.

File: README.md
Comment (ErinUntermeyer) [inline README.md:59]: My vote is for`/api/v1/applicants/search-options` because we are referring to them as options throughout the codebase

**Review by ErinUntermeyer — CHANGES_REQUESTED** [review]
Line 95 and 97 have some copy/paste typos that will need to be fixed before merging in. :) Everything else is great! Thank you for knocking this out Greyson.

File: README.md
Comment (amykarnaze) [inline README.md:59]: I think there should be something to indicate this is a list of total search options? Or, this might be inferred. I prefer search-options.

**Review by amykarnaze — APPROVED** [review]
The endpoints and corresponding data look good. Should the user have any other information to display?

File: README.md
Comment (GreysonElkins) [reply README.md:72]: The request body will be constructed on the FE.
The filtering search will happen on the BE.

**Review by GreysonElkins — COMMENTED** [review]
Nice catch with that messed up user @ErinUntermeyer! Fixed!

File: README.md
Comment (rrabinovitch) [reply README.md:41]: ooh same

**Review by ErinUntermeyer — APPROVED** [review]

File: README.md
Comment (ErinUntermeyer) [reply README.md:72]: Alright so the FE will handle the logic that an employer cannot click "Search" if there are no skills or values selected! Cool, thanks!

File: README.md
Comment (rrabinovitch) [inline README.md]: so @gabichuelas and I are having a hard time figuring out how to include an `updated_at` attribute and are hoping it'd be alright if we plan to come back to this later when time allows, since the filtering by when a profile has been updated isn't exactly essential at this point? we'll still have a `created_at` attribute

File: README.md
Comment (rrabinovitch) [reply README.md:59]: `search-options` makes sense to me (not totally conventional ReST-wise but I'm not sure there's a good restful option)

**Review by rrabinovitch — COMMENTED** [review]
sweeeet thanks for putting this all together - it's helpful to have this as a guide
@gaby and i left a note about `updated_at` - let us know if that's alright with yall

ps. gonna remove Gaby from the review requests since we looked at this together

File: README.md
Comment (ErinUntermeyer) [reply README.md]: I'm fine with putting it on hold for now! @GreysonElkins @amykarnaze what do y'all think?

Comment (GreysonElkins) [reply README.md]: I agree. We were thinking of using this as a part of our sorting for search results (not serving up accounts that have been inactive for 6 months above ones which were updated yesterday, for example) but this is not essential and can definitely happen sometime down the line. 

@ErinUntermeyer do we know if TypeScript will be mad if we leave our definitions as including `updated_at` when we start receiving data that doesn't have that property?

Comment (amykarnaze) [reply README.md]: On hold works for me! This could be tackled when we decide to implement the log in.

Comment (ErinUntermeyer) [reply README.md]: TypeScript will be mad. We will have to match our definitions to the data we get back! But should be a very quick fix if the data changes in the future now that we have made a definitions file and only have to update it in a single place. @GreysonElkins

Comment (amykarnaze) [reply README.md]: Should we omit the updated_at property or set it to null for now? Omitting it until we are able to us it may be the best practice.

Comment (GreysonElkins) [reply README.md]: All those things sound good!

File: README.md
Comment (gabichuelas) [inline README.md:161]: Just wanted to point out that according to regular database model convention, the id for the object in question (in this case a message) will just be called `id` not `message_id`. It's less confusing this way, because when you see a column that specifies what kind of `id` it is, it's usually references a resources on a _different_ table. 

No need to hold up the PR for this, but just a heads' up for FE. I can change this on the next PR if y'all don't get to it. 
@GreysonElkins @rrabinovitch

**Review by gabichuelas — APPROVED** [review]

**Pull request merged**
